### PR TITLE
[MINOR][DOCS] Fix wrong reference

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to fix wrong reference, include:
- Hadoop doc related
- Kafka doc related
- Other

### Why are the changes needed?
Update reference.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manual testing.

### CODE REVIEW

File: connector/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/ConsumerStrategy.scala [hunk 1]
```diff
@@ -39,15 +39,15 @@ import org.apache.spark.kafka010.KafkaConfigUpdater
  */
 abstract class ConsumerStrategy[K, V] {
   /**
-   * Kafka <a href="http://kafka.apache.org/documentation.html#newconsumerconfigs">
+   * Kafka <a href="https://kafka.apache.org/documentation.html#consumerconfigs">
    * configuration parameters</a> to be used on executors. Requires "bootstrap.servers" to be set
```
Comment (panbingkun) [inline connector/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/ConsumerStrategy.scala:43]: The old link has outdated

File: core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala [hunk 2]
```diff
@@ -560,7 +560,7 @@ private[spark] object SparkHadoopUtil extends Logging {
    * Create a file on the given file system, optionally making sure erasure coding is disabled.
    *
    * Disabling EC can be helpful as HDFS EC doesn't support hflush(), hsync(), or append().
-   * https://hadoop.apache.org/docs/r3.0.0/hadoop-project-dist/hadoop-hdfs/HDFSErasureCoding.html#Limitations
```
Comment (panbingkun) [inline core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala:563]: The Hadoop version that Spark relies on has exceeded 3.0

File: streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala [hunk 3]
```diff
@@ -173,7 +173,7 @@ object Checkpoint extends Logging {
       // to find classes, which maybe the wrong class loader. Hence, an inherited version
       // of ObjectInputStream is used to explicitly use the current thread's default class
       // loader to find and load classes. This is a well know Java issue and has popped up
-      // in other places (e.g., http://jira.codehaus.org/browse/GROOVY-1627)
```
Comment (panbingkun) [inline streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala:176]: Dead link

**Review by wangyum — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**